### PR TITLE
ticket 9 done

### PR DESCRIPTION
**Pull request by yaahcd**

### CODE REVIEW

File: src/components/SearchBar.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+			<section className="errorMessage">
+				<p>{error.request.status}: {error.response.data.msg}</p>
+			</section>
+		);
+	}
+
+  if(isLoading){
+    return <Loading />
+  }
+
```
Comment (sirKevlar) [inline src/components/SearchBar.jsx:46]: We should only render below if isLoading is not true

File: src/components/SearchBar.jsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import Loading from "./Loading"
+
+function SearchBar({setArticlesList}) {
+ 
+  const [topicsList, setTopicsList] = useState()
+  const [topicQuery, setTopicQuery] = useState()
+  const [isLoading, setIsLoading] = useState(true)
+	const [error, setError] = useState(false)
+
+  useEffect(() => {
```
Comment (sirKevlar) [inline src/components/SearchBar.jsx:12]: We need to ensure error is set to false every time this useEffect runs

File: src/components/SearchBar.jsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    getTopics().then((topics) => {
+      setTopicsList([{slug: '' }, ...topics])
+      setIsLoading(false)
+    }).catch((err) => {
+      setError(err);
```
Comment (sirKevlar) [inline src/components/SearchBar.jsx:17]: Error should be a boolean, but here we are setting error to be the error object

File: src/components/ListOfArticles.jsx [hunk 4]
```diff
@@ -35,11 +36,14 @@ function ListOfArticles() {
 	}
 
 	return (
+		<>
+		<SearchBar setArticlesList={setArticlesList}/>
 		<ul className="articlesList">
-			{articlesList.map((article) => {
+			{articlesList.length === 0 ? <p>No articles on this topic.</p> : articlesList.map((article) => {
```
Comment (NRMSMITH) [inline src/components/ListOfArticles.jsx:42]: Great consideration for the user! 😄

File: src/components/SearchBar.jsx (hunk 2 above)
Comment (NRMSMITH) [reply src/components/SearchBar.jsx:12]: I concur with Kev - before the API call in your useEffect, you should be setting error to false (even though it is initiated with false - that could change!!)

File: src/components/SearchBar.jsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+  useEffect(() => {
+    getTopics().then((topics) => {
+      setTopicsList([{slug: '' }, ...topics])
+      setIsLoading(false)
+    }).catch((err) => {
+      setError(true);
+      setIsLoading(false);
+    });
+  }, [])
```
Comment (NRMSMITH) [inline src/components/SearchBar.jsx:20]: I would imagine that this would need a change in topic in the dependency array?

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Hi Yara! 👋 
I've left comments for a few things for you to consider - but I've also noticed that you're working with the test data! It's a good idea for you to reseed your database with the development data now - that should make it tonnes easier for you further down the line (and give you way more to show off with!)

**Pull request merged**